122 documentation website mkdocs (#192)
eScienceLab/Cratey-Validator: docs/contribution.md

2. Commit the change in that branch.
3. Push that branch to your fork of this repository on GitHub.
4. Submit a pull request from that branch to the [upstream repository](https://github.com/eScienceLab/RO-Crate-Validation-Service).
5. If you receive feedback, make the changes in your local clone and push to your branch on GitHub: the pull request will update automatically.

!!! warning
    Note that we use the `develop` branch for development work, and this is where your PR should be aimed. The `main` branch is used for releases, and only pull requests from the `develop` branch are accepted to this.

## Development stack

The development Compose file builds the image from the local `Dockerfile` and mounts the repository's test profiles into both the API and worker containers:

```bash
docker compose -f docker-compose-develop.yml up --build
```

Here `--build` matters: without it, Compose reuses the previously built image and local code changes are not picked up. Add `--profile objectstore` to start the bundled RustFS store for storage-backed work; configuration is the same as in [Installation & Setup](installation.md#configuration-reference).

## Tests

> Install the development dependencies, then run the unit tests, which need no Docker:

Author: ```suggestion
Install the development dependencies, then run the unit tests, which do not use Docker Engine:
```

```bash
pip install -r requirements-dev.txt
```

```bash
pytest --ignore=tests/test_integration.py
```

> The integration tests bring up the full Compose stack (including the object store) and seed crates with `boto3`, so they need Docker running:

Author: ```suggestion
The integration tests bring up the full Compose stack (including the object store) and seed crates with `boto3`, for which they need Docker Engine to be running:
```

```bash
pytest tests/test_integration.py
```

`tests/` mirrors the layout of the `app/` package, so the tests for a module are in the matching directory.

## Linting

The project uses [Ruff](https://docs.astral.sh/ruff/) for linting and formatting, configured in `pyproject.toml`:

```bash
ruff check . && ruff format --check .
```

## Dependencies

> Direct dependencies are declared in `pyproject.toml`; the `requirements*.txt` files are locks generated with pip-compile:

Author: ```suggestion
Direct dependencies are declared in `pyproject.toml`; while the `requirements*.txt` files are locks generated using `pip-compile`:
```

```bash
pip-compile pyproject.toml -o requirements.txt
```

```bash
pip-compile --extra dev pyproject.toml -o requirements-dev.txt
```

## Continuous Integration

> Pull requests to `develop` run three workflows: unit tests, integration tests (which start the Compose stack), and lint (`ruff check` and `ruff format --check`).

Author: ```suggestion
Pull requests to `develop` will trigger three workflows: unit tests, integration tests (which start the Compose stack), and lint (`ruff check` and `ruff format --check`).
```

## How the API works

> The API process handles HTTP and runs metadata-only validation inline. Object storage-backed validation is queued through Redis to a Celery worker, which reads the crate from the S3-compatible store, validates it, and writes the result back:

Author: ```suggestion
The API server handles HTTP and runs metadata-only validation inline. Object storage-backed validation is queued through Redis to a Celery worker, which reads the crate from the S3-compatible store, validates it, and writes the result back:
```
